src/devices/Ak8963/Ak8963.cs

```diff
@@ -0,0 +1,275 @@
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.IO;
+using System.Numerics;
+using System.Threading;
+
+namespace Iot.Device.Ak8963
+{
+    /// <summary>
+    /// AK8963 class implementing a magnetometer
+    /// </summary>
+    public class Ak8963 : IDisposable
+    {
+        private I2cDevice _i2cDevice;
+        private bool _autoDispose;
+        private MeasurementMode _measurementMode;
+        private OutputBitMode _outputBitMode;
+        private bool _selfTest = false;
+        private Vector3 _magnetometerBias = Vector3.Zero;
+        private Ak8963Interface _ak8963Interface;
+
+        public const byte DefaultI2cAddress = 0x0C;
+
+        public Ak8963(I2cDevice i2CDevice, bool autoDispose = true) : this(i2CDevice, new Ak8963I2c(), autoDispose)
+        { }
+
+        public Ak8963(I2cDevice i2CDevice, Ak8963Interface ak8963Interface, bool autoDispose = true)
+        {
+            _i2cDevice = i2CDevice;
+            _autoDispose = autoDispose;
+            _ak8963Interface = ak8963Interface;
+            // Initialize the default modes
+            _measurementMode = MeasurementMode.PowerDown;
+            _outputBitMode = OutputBitMode.Output14bit;
+            byte mode = (byte)((byte)_measurementMode | ((byte)_outputBitMode << 4));
+            WriteRegister(Register.CNTL, mode);
+            if (!CheckVersion())
+                throw new IOException($"This device does not contain the correct signature 0x48 for a AK8963");
+        }
+
+        /// <summary>
+        /// Reset the device
+        /// </summary>
+        public void Reset()
+        {
+            WriteRegister(Register.RSV, 0x01);
+            // When powering the AK893, doc says 50 ms needed
+            Thread.Sleep(50);
+        }
+
+        /// <summary>
+        /// Get the device information
+        /// </summary>
+        /// <returns></returns>
+        public byte GetDeviceInfo()
+        {
+            return ReadByte(Register.INFO);
+        }
+
+        /// <summary>
+        /// Get the magnetometer bias
+        /// </summary>
+        public Vector3 MagnometerBias
+        {
+            get { return _magnetometerBias; }
+            set { _magnetometerBias = value; }
+        }
+
+        /// <summary>
+        /// Calibrate the magnetometer. Make sure your sensor is as far as possible of magnet
+        /// Calculate as well the magnetometer bias
+        /// </summary>
+        /// <returns>Returns the factory calibration data</returns>
+        public Vector3 CalibrateMagnetometer()
+        {
+            Vector3 calib = new Vector3();
+            Span<byte> rawData = stackalloc byte[3];
+
+            var oldPower = MeasurementMode;
+
+            // Stop the magetometer
+            MeasurementMode = MeasurementMode.PowerDown;
+            // Enter the magnetometer Fuse mode to read the calibration data
+            MeasurementMode = MeasurementMode.FuseRomAccess;
+            // Read the data
```

Comment:
NIT: can you mention in this comment just the page where this formula was extracted from in the datasheet? We usually do that so that it is easier to go change this binding in the future.